src/main/frontend/src/validators.js

```diff
@@ -25,5 +25,5 @@ export const routeRule = (value) => /^[a-zA-Z_$/?%-]+$/.test(value) ? undefined
 export const xoomVersionRule = (value) => ["1.4.1-SNAPSHOT", "1.4.0"].some(v => v == value) ? undefined : errors.XOOMVERSION;
 export const schemaGroupRule = (value) => /^[A-Za-z]+\:[A-Za-z]+\:[A-Za-z.]+$/.test(value) ? undefined : errors.SCHEMAGROUP;
 export const schemaRule = (value) => /^[A-Za-z]+\:[A-Za-z]+\:[A-Za-z.]+\:[A-Za-z]+\:\d+\.\d+\.\d+$/.test(value) ? undefined : errors.SCHEMA;
-export const isPropertyUnique = (value, store, prop) => store.filter(obj => obj[prop] === value).length === 1 ? undefined : `${prop.charAt(0).toUpperCase() + prop.slice(1)} must be unique`;
-export const isAggregateUnique = (aggregateIndex, aggregateName, store) => store.filter(obj => obj.aggregateIndex != aggregateIndex && obj.aggregateName === aggregateName).length === 1 ? undefined : `Aggregate name must be unique`;
\ No newline at end of file
+export const isPropertyUniqueRule = (value, array, prop) => array.filter(obj => obj[prop] === value).length === 1 ? undefined : `${prop.charAt(0).toUpperCase() + prop.slice(1)} must be unique`;
+export const isAggregateUniqueRule = (oldAggregate, aggregateName, aggregateSettings) => aggregateSettings.filter(a => JSON.stringify(a) !== JSON.stringify(oldAggregate) && a.aggregateName === aggregateName).length === 0 ? undefined : `Aggregate name must be unique`;
```

Comment:
I don't why we ever passed `[...$aggregateSettings, {aggregateName}]` here, also naming changes

Reply:
> https://vlingo-xoom-starter-git-refactor-dialog-florian-schoenherr.vercel.app/
> Try it by going on the navbar.

@Florian-Schoenherr given URL doesn't work as expected can you check it

![image](https://user-images.githubusercontent.com/29024217/111555752-1c090a80-879a-11eb-8d22-5d49a4996fdc.png)